### PR TITLE
ADR-006 stage 2

### PR DESCRIPTION
**Pull request by p0mvn**
This PR implements ADR-006 stage:
https://www.notion.so/osmosiszone/ADR-006-Streaming-GRPC-SQS-Ingest-4e3b2ff7d23e43e2a1f3c43adc3c26bc

It entails removing Redis as a dependency and pushing updates to SQS via GRPC.

As part of this change, poolIngester abstraction was renamed to poolTransformer because it does not ingest data into Redis anymore. Rather, it only performs data transformations on pools by instrumenting TVL data. The new name reflects this.

Config was also updated, replacing Redis parameters with GRPC.

This is a PR Osmosis side: https://github.com/osmosis-labs/osmosis/pull/7952

### CODE REVIEW

File: domain/telemetry.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	// * height - the height of the block being processed
+	SQSIngestUsecaseProcessBlockErrorMetricName = "sqs_ingest_usecase_process_block_error"
+
+	// sqs_ingest_usecase_process_block_error
+	//
+	// counter that measures the number of errors that occur during pool parsing in ingest usecase
+	//
+	// Has the following labels:
+	// * err - the error message occurred
+	SQSIngestUsecaseParsePoolErrorMetricName = "sqs_ingest_usecase_parse_pool_error"
```
Comment (p0mvn) [inline domain/telemetry.go:29]: Note: linking to Linear task: https://linear.app/osmosis/issue/STABI-70/instrument-new-metrics-for-stage-2-sqs-grpc-refactor

File: domain/telemetry.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	// sqs_ingest_usecase_process_block_error
+	//
+	// counter that measures the number of errors that occur during processing a block in ingest usecase
+	//
+	// Has the following labels:
+	// * err - the error message occurred
+	// * height - the height of the block being processed
+	SQSIngestUsecaseProcessBlockErrorMetricName = "sqs_ingest_usecase_process_block_error"
+
+	// sqs_ingest_usecase_process_block_error
```
Comment (p0mvn) [inline domain/telemetry.go:23]:
```suggestion
	// sqs_ingest_usecase_parse_pool_error
```

File: domain/telemetry.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	// * height - the height of the block being processed
+	SQSIngestUsecaseProcessBlockErrorMetricName = "sqs_ingest_usecase_process_block_error"
+
+	// sqs_ingest_usecase_parse_pool_error
+	//
+	// counter that measures the number of errors that occur during pool parsing in ingest usecase
+	//
+	// Has the following labels:
+	// * err - the error message occurred
+	SQSIngestUsecaseParsePoolErrorMetricName = "sqs_ingest_usecase_parse_pool_error"
```
Comment (p0mvn) [inline domain/telemetry.go:29]: Note: linking to Linear task: https://linear.app/osmosis/issue/STABI-70/instrument-new-metrics-for-stage-2-sqs-grpc-refactor

File: domain/telemetry.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	// * height - the height of the block being processed
+	SQSIngestUsecaseProcessBlockErrorMetricName = "sqs_ingest_usecase_process_block_error_total"
+
+	// sqs_ingest_usecase_parse_pool_error_total
+	//
+	// counter that measures the number of errors that occur during pool parsing in ingest usecase
+	//
+	// Has the following labels:
+	// * err - the error message occurred
+	SQSIngestUsecaseParsePoolErrorMetricName = "sqs_ingest_usecase_parse_pool_error_total"
```
Comment (p0mvn) [inline domain/telemetry.go:29]: Note: linking to Linear task: https://linear.app/osmosis/issue/STABI-70/instrument-new-metrics-for-stage-2-sqs-grpc-refactor

**Review by PaddyMc — APPROVED** [review]

**Pull request merged**